generator/src/lib.rs

```diff
@@ -2085,22 +2082,14 @@ pub fn generate_feature<'a>(
     let (static_commands, entry_commands, device_commands, instance_commands) = feature
         .elements
         .iter()
-        .flat_map(|feature| {
-            if let vkxml::FeatureElement::Require(ref spec) = feature {
-                spec.elements
-                    .iter()
-                    .filter_map(|feature_spec| {
-                        if let vkxml::FeatureReference::CommandReference(ref cmd_ref) = feature_spec
-                        {
-                            Some(cmd_ref)
-                        } else {
-                            None
-                        }
-                    })
-                    .collect()
-            } else {
-                vec![]
-            }
+        .filter_map(|feature| match feature {
+            vkxml::FeatureElement::Require(ref spec) => Some(spec),
+            _ => None,
+        })
+        .flat_map(|spec| spec.elements.iter())
+        .filter_map(|feature_spec| match feature_spec {
+            vkxml::FeatureReference::CommandReference(ref cmd_ref) => Some(cmd_ref),
+            _ => None,
         })
         .filter_map(|cmd_ref| commands.get(&cmd_ref.name))
```

Comment:
Let's not have two `filter_map`s chained in a row.

Reply:
Part of this is that it sort of separated out a repeated pattern of `match x { X::Foo(y) => y, _ => None }`, which ended up making it clearer to me that a `get_variant` macro would be useful. Would it be preferable to avoid this intermediate commit?